lib/rubocop/cop/style/and_or.rb

```diff
@@ -63,25 +63,47 @@ def correction(node)
           replacement = (node.type == :and ? '&&' : '||')
           lambda do |corrector|
             [expr1, expr2].each do |expr|
-              _receiver, _method_name, *args = *expr
-              next unless correctable?(expr)
-
-              sb = expr.loc.expression.source_buffer
-              begin_paren = expr.loc.selector.end_pos
-              end_paren = begin_paren + 1
-              range = Parser::Source::Range.new(sb, begin_paren, end_paren)
-              corrector.replace(range, '(')
-              corrector.insert_after(args.last.loc.expression, ')')
+              if expr.send_type?
+                correct_send(expr, corrector)
+              elsif expr.return_type?
+                correct_other(expr, corrector)
+              elsif ASGN_NODES.include?(expr.type)
+                correct_other(expr, corrector)
+              end
             end
             corrector.replace(node.loc.operator, replacement)
           end
         end
 
-        def correctable?(expr)
-          return false unless expr.type == :send
-          _receiver, method_name, *args = *expr
+        def correct_send(node, corrector)
+          receiver, method_name, *args = *node
+          if method_name == :!
+            # ! is a special case:
+            # 'x and !obj.method arg' can be auto-corrected if we
+            # recurse down a level and add parens to 'obj.method arg'
+            # I haven't found any other operator for which this is necessary
```

Comment:
Use passive voice rather than referring to yourself, e.g. "Any other operator for which this is necessary has not been found."